lib/galaxy/datatypes/binary.py

```diff
@@ -2013,6 +2013,69 @@ def display_peek(self, dataset):
             return "SearchGUI Archive, version %s" % (dataset.metadata.searchgui_version or 'unknown')
 
 
+class WiffArchive(CompressedArchive):
+    """Class describing a AB SCIEX archive of wiff format files"""
+    MetadataElement(name="wiff_file", default=None, param=MetadataParameter, desc="Wiff File Path",
+                    readonly=True, visible=True, no_value=None)
+    MetadataElement(name="wiff_files", default=[], param=MetadataParameter, desc="Wiff Files",
+                    readonly=True, visible=False, no_value=None)
+    file_ext = "wiff_archive"
+
+    def set_meta(self, dataset, overwrite=True, **kwd):
+        super(WiffArchive, self).set_meta(dataset, overwrite=overwrite, **kwd)
+        try:
+            if dataset and zipfile.is_zipfile(dataset.file_name):
+                with zipfile.ZipFile(dataset.file_name) as tempzip:
+                    wiff_files = []
+                    for name in tempzip.namelist():
+                        if name.lower().endswith('.wiff'):
+                            dataset.metadata.wiff_file = name
+                            wiff_files.append(name)
+                        elif name.lower().find('.wiff.') > 0:
+                            wiff_files.append(name)
+                    dataset.metadata.wiff_files = sorted(wiff_files)
+        except Exception as e:
+            log.warning('%s, set_meta Exception: %s', self, e)
+
+    def sniff(self, filename):
+        """
+        Try to guess if the file is a WiffArchive.
+
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname('wiff.zip')
+        >>> WiffArchive().sniff(fname)
+        True
+
+        >>> fname = get_test_fname('drugbank_drugs.mz5')
+        >>> WiffArchive().sniff(fname)
+        False
+        """
+        try:
+            if filename and zipfile.is_zipfile(filename):
+                with zipfile.ZipFile(filename, 'r') as tempzip:
+                    for name in tempzip.namelist():
+                        if name.lower().endswith('.wiff'):
+                            return True
+                return False
+        except Exception as e:
```

Comment:
It's not necessary to do this, exceptions are caught automatically, and I don't think we want to see failed sniffing attempts in the logs, as that's a very normal thing.